app/frontend/courses/review/components/CoursesReview__Editor.res

```diff
@@ -1194,6 +1194,23 @@ let make = (
     },
   )
 
+  React.useEffect1(() => {
+    let handleBeforeUnload = event => {
+      if state.newFeedback != "" {
+        event |> Webapi.Dom.Event.preventDefault
+        DomUtils.Event.setReturnValue(event, "")
+      }
+    }
+
+    Webapi.Dom.Window.addEventListener("beforeunload", handleBeforeUnload, Webapi.Dom.window)
+
+    let removeEventListener = () => {
+      Webapi.Dom.Window.removeEventListener("beforeunload", handleBeforeUnload, Webapi.Dom.window)
+    }
+
+    Some(removeEventListener)
```

Comment:
This process of setting up a listener and tearing it down each time the feedback changes feels quite wasteful. I think it'd be much more performant to set up the listener once, and then use a ref to make sure that the listener has access to the latest  `newFeedback` value. Something like this:

```res
let newFeedbackRef = React.useRef(state.newFeedback)

React.useEffect0(() => {
  let handleBeforeUnload = event => {
    %debugger
    if newFeedbackRef.current != "" {
      Webapi.Dom.Event.preventDefault(event)
      DomUtils.Event.setReturnValue(event, "")
    }
  }

  Webapi.Dom.Window.addEventListener("beforeunload", handleBeforeUnload, Webapi.Dom.window)

  let removeEventListener = () => {
    Webapi.Dom.Window.removeEventListener("beforeunload", handleBeforeUnload, Webapi.Dom.window)
  }

  Some(removeEventListener)
})

React.useEffect1(() => {
  newFeedbackRef.current = state.newFeedback
  None
}, [state.newFeedback])
```

This way, the op that runs every time feedback is changed is as simple as it can be (changing the ref), and the listener is set up only once (component mount) and torn down only once (component unmount).